### PR TITLE
chore: assertions package

### PR DESCRIPTION
**Pull request by mxkaske**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation

### Before this PR

{Please add a screenshot here}

### After this PR

{Please add a screenshot here}

### CODE REVIEW

File: apps/checker/cmd/main.go [hunk 1]
```diff
@@ -93,12 +103,58 @@ func main() {
...
+						fmt.Println("⚠️ Not Handled assertion type", assert.AssertionType)
+					}
+				}
+			} else {
+				isSuccessfull = statusCode.IsSuccessful()
+			}
+
 			// let's retry at least once if the status code is not successful.
-			if !statusCode.IsSuccessful() && called < 2 {
+			if !isSuccessfull && called < 2 {
```
Comment (mxkaske) [inline apps/checker/cmd/main.go:145]: Should we **only** consider retry when status code is wrong, or the whole assertions?

File: apps/checker/cmd/main.go [hunk 2]
```diff
@@ -136,6 +193,8 @@ func main() {
 				Timestamp:     req.CronTimestamp,
 				MonitorID:     req.MonitorID,
 				WorkspaceID:   req.WorkspaceID,
+				Error:         1,
```
Comment (mxkaske) [inline apps/checker/cmd/main.go:196]: Are you sure about that?

File: apps/checker/cmd/main.go (hunk 1 above)
Comment (thibaultleouay) [reply apps/checker/cmd/main.go:145]: Whole assertion

To avoid false positive

File: apps/checker/cmd/main.go (hunk 2 above)
Comment (thibaultleouay) [reply apps/checker/cmd/main.go:196]: Yep this happens when we have an error at the third retry

**Pull request merged**